src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx

```diff
@@ -78,48 +80,94 @@ export default class IdvActivityTooltip extends React.Component {
+    }
 
-    truncateText(awardingAgencyNameDiv, recipientNameDiv, theTooltipWidth) {
+    truncateLogic(arrayOfDivsToTruncate, theTooltipWidth) {
         return new Promise((resolve) => {
-            const { graphWidth, recipientName, awardingAgencyName } = this.props.data;
-            // get awardingAgencyName truncated width based on the tooltip width and graph width
-            const truncatedTextWidthAgency =
-            (graphWidth * awardingAgencyNameDiv) / theTooltipWidth;
-            // get recipientName truncated width based on the tooltip width and graph width
-            const truncatedTextWidthRecipient =
-            (graphWidth * recipientNameDiv) / theTooltipWidth;
-            // we now have a ratio of the truncated width to the total width,
-            // and we have the total length
-            // we can now find the appropriate truncated
-            // length for the awardingAgencyName and recipientName
-            const truncatedTextLengthAgency = Math.floor(
-                ((truncatedTextWidthAgency * awardingAgencyName.length) / awardingAgencyNameDiv) / 2);
-            const truncatedTextLengthRecipient = Math.floor(
-                ((truncatedTextWidthRecipient * recipientName.length) / recipientNameDiv) / 2);
-            const truncatedAgency = `${awardingAgencyName
-                .substring(0, truncatedTextLengthAgency).trim()}...`;
-            const truncatedRecipient = `${recipientName
-                .substring(0, truncatedTextLengthRecipient).trim()}...`;
-            this.setState({
-                awardingAgency: truncatedAgency,
-                recipient: truncatedRecipient,
-                truncated: true
-            }, () => resolve());
+            const {
+                graphWidth,
+                recipientName,
+                awardingAgencyName,
+                piid,
+                parentAwardPIID
+            } = this.props.data;
+            // need this to find the appropriate text to pass to
+            // truncatedCharacterLength
+            const arrayOfDivText = [
+                recipientName,
+                awardingAgencyName,
+                piid,
+                parentAwardPIID
+            ];
+            // need this to set the properties for state
+            const arrayOfProperties = [
+                'recipientName',
+                'awardingAgencyName',
+                'piid',
+                'parentAwardPIID'
+            ];
+            const truncatedWidths = arrayOfDivsToTruncate.map((divWidth) => this.truncatedWidth(
+                graphWidth, theTooltipWidth, divWidth));
+            const truncatedLengths = truncatedWidths
+                .map((truncatedDivWidth, index) => this.truncatedCharacterLength(
+                    truncatedDivWidth,
+                    arrayOfDivsToTruncate[index],
+                    arrayOfDivText[index].length));
+            const truncatedText = truncatedLengths.reduce((acc, truncatedLength, index) => {
```

Comment:
Instead of declaring three variables here (`line 130, 132 & 137`) I would recommend chaining as follows:

```javascript
const truncatedText = arrayOfDivsToTruncate.map().map().reduce()
```

Also maybe think of a more descriptive name for this single variable like `truncatedStatePropertiesObj` or something?